qiita_pet/templates/analysis_selected.html

```diff
@@ -1,8 +1,15 @@
 {% extends sitebase.html %}
 {% block head %}
 <script type="text/javascript">
+var ws = init_websocket(window.location.host + '/analysis/selected/socket/');
+ws.onmessage = function(evt) { if(evt.data != 'true') {ws.close();} };
+ws.onerror = function(evt) { $('#ws-error').html("<b>Server communication error. Sample removal will not be recorded. Please try again later.</b>"); };
+ws.onclose = function(evt) { $('#ws-error').html("<b>Server communication error. Sample removal will not be recorded. Please try again later.</b>"); };
```

Comment:
n this case, it should stay open until you leave the page in order to capture sample removals. If it closes prematurely that’s bad and should be treated as an error.
> On Apr 22, 2015, at 7:20 AM, Daniel McDonald <notifications@github.com> wrote:
> 
> In qiita_pet/templates/analysis_selected.html <https://github.com/biocore/qiita/pull/1058#discussion_r28876200>:
> 
> > @@ -1,8 +1,15 @@
> >  {% extends sitebase.html %}
> >  {% block head %}
> >  <script type="text/javascript">
> > +var ws = init_websocket(window.location.host + '/analysis/selected/socket/');
> > +ws.onmessage = function(evt) { if(evt.data != 'true') {ws.close();} };
> > +ws.onerror = function(evt) { $('#ws-error').html("<b>Server communication error. Sample removal will not be recorded. Please try again later.</b>"); };
> > +ws.onclose = function(evt) { $('#ws-error').html("<b>Server communication error. Sample removal will not be recorded. Please try again later.</b>"); };
> Closing a web socket should be a normal operation, right? Why is this considered an error?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/biocore/qiita/pull/1058/files#r28876200>.
>